galsim/utilities.py

```diff
@@ -1913,3 +1913,244 @@ def __exit__(self, type, value, traceback):
         set_omp_threads(self.orig_num_threads)
         if self.tpl is not None:  # pragma: no cover
             self.tpl.unregister()
+
+
+
+# The rest of these are only used by the tests in GalSim.  But we make them available
+# for other code bases who may want to use them as well.
+
+def do_pickle(obj, func = lambda x : x, irreprable=False, random=None):
+    """Check that the object is picklable.
+
+    Also check some related desirable properties that we want for (almost) all galsim objects:
+
+    1. pickle.loads(pickle.dumps(obj)) recovers something equivalent to the original.
+    2. obj != object() and object() != obj.  (I.e. it doesn't == things it shouldn't.)
+    3. hash(obj) is the same for two equivalent objects, if it is hashable.
+    4. copy.copy(obj) returns an equivalent copy unless random=True.
+    5. copy.deepcopy(obj) returns an equivalent copy.
+    6. eval(repr(obj)) returns an equivalent copy unless random or irreprable=True.
+    7. repr(obj) makes something if irreprable=False.
+
+    Parameters:
+        obj:            The object to test
+        func:           A function of obj to use to test for equivalence.  [default: lambda x: x]
+        irreprable:     Whether to skip the eval(repr(obj)) test.  [default: False]
+        random:         Whether the obj has some intrinsic randomness. [default: False, unless
```

Comment:
Looks like default is `None`?

Reply:
Right.  I was describing what that default means, which is False normally, but tries to notice randomness.